vm/src/stdlib/syslog.rs

```diff
@@ -0,0 +1,132 @@
+pub(crate) use syslog::make_module;
+
+#[pymodule(name = "syslog")]
+mod syslog {
+    use crate::builtins::pystr::{PyStr, PyStrRef};
+    use crate::common::lock::PyRwLock;
+    use crate::function::{OptionalArg, OptionalOption};
+    use crate::pyobject::PyValue;
+    use crate::utils::ToCString;
+    use crate::vm::VirtualMachine;
+    use crate::{PyObjectRef, PyResult, TryFromObject};
+    use std::ffi::CStr;
+
+    #[pyattr]
+    use libc::{
+        LOG_ALERT, LOG_AUTH, LOG_CONS, LOG_CRIT, LOG_DAEMON, LOG_DEBUG, LOG_EMERG, LOG_ERR,
+        LOG_INFO, LOG_KERN, LOG_LOCAL0, LOG_LOCAL1, LOG_LOCAL2, LOG_LOCAL3, LOG_LOCAL4, LOG_LOCAL5,
+        LOG_LOCAL6, LOG_LOCAL7, LOG_LPR, LOG_MAIL, LOG_NDELAY, LOG_NEWS, LOG_NOTICE, LOG_NOWAIT,
+        LOG_ODELAY, LOG_PID, LOG_SYSLOG, LOG_USER, LOG_UUCP, LOG_WARNING,
+    };
+
+    #[cfg(not(target_os = "redox"))]
+    #[pyattr]
+    use libc::{LOG_AUTHPRIV, LOG_CRON, LOG_PERROR};
+
+    fn get_argv(vm: &VirtualMachine) -> Option<PyStrRef> {
+        if let Some(argv) = vm.state.settings.argv.first() {
+            if !argv.is_empty() {
+                return Some(
+                    PyStr::from(match argv.find('\\') {
+                        Some(value) => &argv[value..],
+                        None => argv,
+                    })
+                    .into_ref(vm),
+                );
+            }
+        }
+        None
+    }
+
+    fn syslog_global_ident() -> &'static PyRwLock<Option<Box<CStr>>> {
+        rustpython_common::static_cell! {
+            static SYSLOG_IDENT: PyRwLock<Option<Box<CStr>>>;
+        };
+        SYSLOG_IDENT.get_or_init(|| PyRwLock::new(None))
+    }
+
+    #[derive(Default, FromArgs)]
+    struct OpenLogArgs {
+        #[pyarg(any, optional)]
+        ident: OptionalOption<PyStrRef>,
+        #[pyarg(any, optional)]
+        logoption: OptionalArg<i32>,
+        #[pyarg(any, optional)]
+        facility: OptionalArg<i32>,
+    }
+
+    #[pyfunction]
+    fn openlog(args: OpenLogArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let logoption = args.logoption.unwrap_or(0);
+        let facility = args.facility.unwrap_or(LOG_USER);
+        let ident = match args.ident.flatten() {
+            Some(args) => {
+                let cstr = args.to_cstring(vm)?.into_boxed_c_str();
+                let cstr_ptr = cstr.as_ptr();
+                *syslog_global_ident().write() = Some(cstr);
+                cstr_ptr
+            }
+            None => match get_argv(vm) {
+                Some(argv) => {
+                    let cstr = argv.to_cstring(vm)?.into_boxed_c_str();
+                    let cstr_ptr = cstr.as_ptr();
+                    *syslog_global_ident().write() = Some(cstr);
+                    cstr_ptr
+                }
+                None => std::ptr::null(),
```

Comment:
no update for `syslog_global_ident()` if this is None? what if openlog is called twice first with args but second without it?

Reply:
Though [the docs](https://www.gnu.org/software/libc/manual/html_node/openlog.html) seem to _indicate_ this should be fine, [CPython sets it to `NULL`](https://github.com/python/cpython/blob/80d9ff16483b6c1898bcdcc811b5450b57a5e573/Modules/syslogmodule.c#L135) so I think we should too.